CMakeLists.txt

```diff
@@ -77,7 +77,7 @@ include_directories(${SFML_INCLUDE_DIR} ${OPENGL_INCLUDE_DIRS} ${GLEW_INCLUDE_DI
 add_executable(Fly ${SOURCES})
 target_link_libraries(Fly ${SFML_LIBRARIES} ${SFML_DEPENDENCIES} ${OPENGL_LIBRARIES} ${GLEW_LIBRARIES})
 
-set_property(TARGET Fly PROPERTY CXX_STANDARD 14)
+set_property(TARGET Fly PROPERTY CXX_STANDARD 17)
```

Comment:
Thanks for PR!
Any reason for the unrelated patches about on Controller.cpp and the C++ version requirement?

Reply:
They are related, it wouldn't compile with auto x = thatfunction so I just replaced every instance of x with thatfunction, as for the version change all the warnings went away after changing it to c++17, 
for context I am not a c++ programmer, I just wanted to get the program to build and also cleared the warnings while I was at it.